### PR TITLE
Add async search in 2.5 manifest

### PR DESCRIPTION
**Pull request by Arpit-Bandejiya**
Signed-off-by: Arpit Bandejiya <abandeji@amazon.com>

### Description
Add async search to 2.5 manifest

### Issues Resolved
Part of [#217](https://github.com/opensearch-project/asynchronous-search/issues/217)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: manifests/2.5.0/opensearch-2.5.0.yml [hunk 1]
```diff
@@ -163,3 +163,12 @@ components:
       - windows
     checks:
       - gradle:properties:version
+  - name: asynchronous-search
+    repository: https://github.com/opensearch-project/asynchronous-search.git
+    ref: 2.5
```
Comment (bharath-techie) [inline manifests/2.5.0/opensearch-2.5.0.yml:168]: Should we add tags or branch in ref ? otherwise approving it

**Review by bharath-techie — APPROVED** [review]

File: manifests/2.5.0/opensearch-2.5.0.yml (hunk 1 above)
Comment (gaiksaya) [reply manifests/2.5.0/opensearch-2.5.0.yml:168]: Branch as tags are cut after the release. @Arpit-Bandejiya please make `2.5` a string by enclosing it in quotes. Thanks!

Comment (Arpit-Bandejiya) [reply manifests/2.5.0/opensearch-2.5.0.yml]: Done @gaiksaya, please check.

Comment (gaiksaya) [reply manifests/2.5.0/opensearch-2.5.0.yml]: Looks like there is a merge conflict. Can you rebase and fix it?
Thanks!

**Review by gaiksaya — APPROVED** [review]

File: manifests/2.5.0/opensearch-2.5.0.yml (hunk 1 above)
Comment (Arpit-Bandejiya) [reply manifests/2.5.0/opensearch-2.5.0.yml]: fixed!

**Comment by gaiksaya** [comment]
Hi @Arpit-Bandejiya Can you rebase? The CI was fixed in recent PR.
Thanks!

**Comment by peterzhuamazon** [comment]
I rebased for @Arpit-Bandejiya let us see if it pass.
Thanks.

**Review by zelinh — APPROVED** [review]

**Comment by zelinh** [comment]
CI are passing. I'm merging this now.

**Pull request merged**